# Bienvenid@s!

**Esta celda de texto es un Markdown. Es el tipo de celda donde podemos escribir texto que no es código.**

Comencemos de inmediato tratando con algunos datos dentro de este Jupyter Notebook. La primera parte de código que ejecutará se encuentra en la celda de abajo. Puede presionar el botón *play* o presionar de manera más eficiente *shift + enter* en su teclado para ejecutar el código.

In [1]:
#This cell is a code cell. It is where we can type code that can be executed.
#The hashtag at the start of this line makes it so that this text is a comment not code. 

import pandas as pd

La razón por la que ejecutamos el código ```import pandas as pd``` es para poder usar las funciones de la libreria ```pandas```, que proporciona estructuras de datos y herramientas de análisis de datos que son realmente útiles. Una de las cosas fantásticas de realizar análisis de datos en Python es la disponibilidad de excelentes herramientas de análisis como ```pandas```. Algo frustrante puede ser aprender a usar estas diversas herramientas y saber cuando usarlas. Te sentirás cada vez más cómodo con estas herramientas a medida que avanza el semestre.

# Buscando tu cumplemoto

Tu cumple-moto o terremoto de nacimiento es el terremoto de mayor magnitud que ocurrió el día en que naciste. En este ejercicio en clase buscaremos un catálogo de terremotos y encontraremos tu cumple-moto.

Para hacerlo, podemos descargar datos del programa de peligro sísmico del Servicio Geológico de EE.UU. (USGS). https://earthquake.usgs.gov

Vamos a utilizar una API (*Application programming interface*
) que nos permite enviar una URL a la USGS y obtener información sobre terremotos para un conjunto de parámetros especificados.

## Buscando el Cumplemoto de Rodrigo


Hagámoslo primero para mi cumpleaños. Definiremos mi cumpleaños en formato año-mes-día y el día después de mi cumpleaños en formato año-mes-día para crear una URL que obtenga datos a partir de las 12 AM de mi cumpleaños y terminando a las 12 AM del día siguiente. Estamos poniendo las comillas ('') alrededor de las fechas para que sean *strings* (el tipo de datos de Python que es una secuencia de texto).

In [2]:
Rodrigo_birthday = '1989-08-18'
day_after_Rodrigo_birthday = '1989-08-19'

Lo que acabamos de hacer en el código anterior es definir la variable ```Rodrigo_birthday``` para que tenga el valor string de ''. Si ejecutamos una celda de código solo con esa variable, Jupyter mostrará la variable como salida del código.

In [3]:
Rodrigo_birthday

'1989-08-18'

Otra forma de ver la variable es decirle a Python que la imprima usando la función `` print() ``.

In [4]:
print(Rodrigo_birthday)

1989-08-18


### Definición de la URL del terremoto del cumpleaños de Rodrigo

Para crear una URL que podamos enviar al USGS (y recuperar datos) necesitamos insertar estas fechas en el formato de URL de la API de terremotos del USGS. Definiremos standard_url como un string y luego agregaremos las fechas que se establecieron arriba para ser la hora de inicio y la hora de finalización.

In [5]:
standard_url = 'https://earthquake.usgs.gov/fdsnws/event/1/query?format=csv&orderby=magnitude'

Rodrigo_birthquake_url = standard_url + '&starttime=' + Rodrigo_birthday + '&endtime=' + day_after_Rodrigo_birthday

Rodrigo_birthquake_url

'https://earthquake.usgs.gov/fdsnws/event/1/query?format=csv&orderby=magnitude&starttime=1989-08-18&endtime=1989-08-19'

### Conseguir los terremotos del cumpleaños de Rodrigo

Ahora tenemos una URL que podemos usar para obtener datos del USGS. Podemos usar las funciones `pandas` que importamos en la parte superior de este notebook para obtener estos datos. La forma estándar de usar ```pandas``` es importarlo con la abreviatura ```pd```. Usaremos la función ```pd.read_csv()``` que tomará los datos que provienen de la URL de USGS y los convertirá en un DataFrame. Un DataFrame es una estructura de datos con columnas de diferentes datos que se definen con los nombres de sus columnas.

In [6]:
Rodrigo_birthday_earthquakes = pd.read_csv(Rodrigo_birthquake_url)

Estos datos están ordenados por magnitud con los terremotos de mayor magnitud en la parte superior. Veamos las primeras 5 filas del DataFrame usando la función ```.head ()```.

In [7]:
Rodrigo_birthday_earthquakes.head()

time  latitude  longitude  depth  mag magType  nst  \
0  1989-08-18T03:46:26.020Z   -55.053    -27.846   33.0  5.6      mw  NaN   
1  1989-08-18T00:14:32.540Z     0.709    126.221   58.9  5.0      mb  NaN   
2  1989-08-18T20:00:51.100Z    51.380    175.649   33.0  4.9      mb  NaN   
3  1989-08-18T09:29:17.390Z    44.446    147.120   33.0  4.8      mb  NaN   
4  1989-08-18T07:43:23.400Z    36.319     71.026  104.9  4.6      mb  NaN   

   gap  dmin  rms  ...                   updated  \
0  NaN   NaN  1.2  ...  2016-11-09T23:16:34.755Z   
1  NaN   NaN  1.0  ...  2014-11-07T00:44:43.430Z   
2  NaN   NaN  1.0  ...  2014-11-07T00:44:43.492Z   
3  NaN   NaN  0.8  ...  2014-11-07T00:44:43.468Z   
4  NaN   NaN  0.8  ...  2014-11-07T00:44:43.454Z   

                                place        type horizontalError depthError  \
0       South Sandwich Islands region  earthquake             NaN        NaN   
1      129 km W of Ternate, Indonesia  earthquake             NaN       11.1   
2   234 km SE of Attu Station, Alaska  earthquake             NaN        NaN   
3       78 km NNE of Shikotan, Russia  earthquake             NaN        NaN   
4  60 km SW of Ashk?sham, Afghanistan  earthquake             NaN       14.4   

   magError  magNst    status  locationSource magSource  
0       NaN     NaN  reviewed              us       hrv  
1       NaN     9.0  reviewed              us        us  
2       NaN    31.0  reviewed              us        us  
3       NaN     5.0  reviewed              us        us  
4       NaN     4.0  reviewed              us        us  

[5 rows x 22 columns]

#### El cumplemoto de Rodrigo.

El terremoto más grande del día que nací fue una magnitud (columna `mag`) 5.6 que ocurrió en el border entre Myanmar e India (columna `place`).

## Encontrando tu cumplemoto
### Ingresa tu cumpleaños

Ingrese su cumpleaños y la fecha siguiente en formato año-mm-dd.

In [8]:
### Enter your birthday here within the quotes in year-mm-dd format

your_birthday = '2002-02-22'
day_after_your_birthday = '2002-02-23'

#### Definición de la URL del terremoto de tu cumpleaños

In [9]:
standard_url = 'https://earthquake.usgs.gov/fdsnws/event/1/query?format=csv&orderby=magnitude'

your_birthquake_url = standard_url + '&starttime=' + your_birthday + '&endtime=' + day_after_your_birthday

your_birthquake_url

'https://earthquake.usgs.gov/fdsnws/event/1/query?format=csv&orderby=magnitude&starttime=2002-02-22&endtime=2002-02-23'

#### Obteniendo los terremotos el día de tu cumpleaños



In [10]:
your_birthday_earthquakes = pd.read_csv(your_birthquake_url)

### Buscando en los terremotos del día de tu cumpleaños

In [11]:
your_birthday_earthquakes.head()

time   latitude  longitude    depth  mag magType  \
0  2002-02-22T19:32:41.750Z  32.318833  -115.3215    7.007  5.7      mw   
1  2002-02-22T23:10:56.250Z -25.510000   -70.1200   48.400  5.4     mwc   
2  2002-02-22T13:52:02.850Z  -6.174000   105.0860   16.600  4.9      mb   
3  2002-02-22T19:30:51.860Z -20.944000  -176.3120  188.700  4.9      mb   
4  2002-02-22T20:04:47.370Z -31.961000    57.3430   10.000  4.9      mb   

     nst    gap    dmin   rms  ...                   updated  \
0   20.0  217.5  0.4751  0.38  ...  2020-07-09T21:11:59.999Z   
1  118.0    NaN     NaN  0.97  ...  2016-11-09T21:56:08.544Z   
2   49.0    NaN     NaN  1.18  ...  2014-11-07T01:14:57.213Z   
3   53.0    NaN     NaN  0.96  ...  2014-11-07T01:14:57.261Z   
4   23.0    NaN     NaN  1.20  ...  2014-11-07T01:14:57.300Z   

                                    place        type horizontalError  \
0             13km WSW of Delta, B.C., MX  earthquake            2.44   
1              38 km ESE of Taltal, Chile  earthquake             NaN   
2  84 km SSW of Bandar Lampung, Indonesia  earthquake             NaN   
3         116 km WNW of Haveluloto, Tonga  earthquake             NaN   
4                  Southwest Indian Ridge  earthquake             NaN   

  depthError  magError  magNst    status  locationSource magSource  
0      31.95       NaN     0.0  reviewed              ci        ci  
1        NaN       NaN     NaN  reviewed              us       hrv  
2        NaN       NaN    11.0  reviewed              us        us  
3        NaN       NaN    24.0  reviewed              us        us  
4        NaN       NaN     7.0  reviewed              us        us  

[5 rows x 22 columns]

Cuál es la magnitud de tu cumple-moto? Llena el siguiente formulario: https://forms.gle/XRVCyFBmW9aFYs3XA

Magnitud 5.7 :o